diff -up ./proc/ksym.c.ori ./proc/ksym.c
--- ./proc/ksym.c.ori	2006-06-25 08:57:18.000000000 +0200
+++ ./proc/ksym.c	2016-11-16 13:34:31.902836748 +0100
@@ -567,11 +567,7 @@ static const char * read_wchan_file(unsi
 
   // lame ppc64 has a '.' in front of every name
   if(*ret=='.') ret++;
-  switch(*ret){
-    case 's': if(!strncmp(ret, "sys_", 4)) ret += 4;   break;
-    case 'd': if(!strncmp(ret, "do_",  3)) ret += 3;   break;
-    case '_': while(*ret=='_') ret++;                  break;
-  }
+  while(*ret=='_') ret++; 
   return ret;
 }
 
@@ -616,11 +612,7 @@ const char * lookup_wchan(unsigned KLONG
   ret = good_symb->name;
   // lame ppc64 has a '.' in front of every name
   if(*ret=='.') ret++;
-  switch(*ret){
-    case 's': if(!strncmp(ret, "sys_", 4)) ret += 4;   break;
-    case 'd': if(!strncmp(ret, "do_",  3)) ret += 3;   break;
-    case '_': while(*ret=='_') ret++;                  break;
-  }
+  while(*ret=='_') ret++;
   /* if(!*ret) ret = fail.name; */  /* not likely (name was "sys_", etc.) */
 
   /* cache name after abbreviation */
