diff -Naur procps-ng-3.3.10.orig/free.c procps-ng-3.3.10/free.c
--- procps-ng-3.3.10.orig/free.c	2015-12-01 17:23:26.702968686 +0100
+++ procps-ng-3.3.10/free.c	2015-12-01 17:22:34.590529390 +0100
@@ -262,6 +262,7 @@
 			break;
 		case 's':
 			flags |= FREE_REPEAT;
+			errno = 0;
 			args.repeat_interval = (1000000 * strtof(optarg, &endptr));
 			if (errno || optarg == endptr || (endptr && *endptr))
 				xerrx(EXIT_FAILURE, _("seconds argument `%s' failed"), optarg);
