diff --git a/proc/sysinfo.c b/proc/sysinfo.c
index baa2453..c95f378 100644
--- a/proc/sysinfo.c
+++ b/proc/sysinfo.c
@@ -979,7 +979,7 @@ unsigned int getpartitions_num(struct disk_stat *disks, int ndisks){
 /////////////////////////////////////////////////////////////////////////////
 static int is_disk(char *dev)
 {
-  char syspath[32];
+  char syspath[64];
   char *slash;
 
   while ((slash = strchr(dev, '/')))
